tests/test_templates_subroutines.py

```diff
@@ -0,0 +1,326 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :mod:`pennylane.template.layers` module.
+"""
+# pylint: disable=protected-access,cell-var-from-loop
+import pytest
+
+import logging as log
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.qnode import QuantumFunctionError
+from pennylane.templates.layers import (Interferometer)
+log.getLogger('defaults')
+
+
+class TestInterferometer:
+    """Tests for the Interferometer from the pennylane.template.layers module."""
+
+    def test_exceptions(self):
+        """test that exceptions are correctly raised"""
+        dev = qml.device('default.gaussian', wires=1)
+        varphi = [0.42342]
+
+        @qml.qnode(dev)
+        def circuit(varphi, mesh):
+            Interferometer(theta=None, phi=None, varphi=varphi, mesh=mesh, wires=0)
+            return qml.expval(qml.NumberOperator(0))
+
+        with pytest.raises(QuantumFunctionError) as excinfo:
+            circuit(varphi, 'rectangular')
+        assert excinfo.value.args[0] == 'The mesh parameter influences the circuit architecture ' \
+                                        'and can not be passed as a QNode parameter.'
```

Comment:
You can use `pytest.raises(QuantumFunctionError, match="can not be passed"):`, which avoids the additional line, and allows you to use regex to match the string instead of matching the exact string.